nsqd/protocol_v2.go

```diff
@@ -113,6 +113,9 @@ func (p *protocolV2) IOLoop(conn net.Conn) error {
 	if client.Channel != nil {
 		client.Channel.RemoveClient(client.ID)
 	}
+	if err == io.EOF && atomic.LoadInt32(&client.State) == stateClosing {
```

Comment:
Ignoring all errors would mask if a connection issued CLS followed by FIN or other messages (which error with `cannot XYZ in current state`). Also, it looks like PUB/MPUB can still be issued while closing.